### ISSUE TITLE
Oracle 19 Alter Table with modify column Parsing Errors

### ISSUE THREAD

**Issue by sriocreux** [sqlfluff/sqlfluff#4815]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

SQLFluff fails to parse statement to alter table and modify column with Oracle dialect

https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/ALTER-TABLE.html#GUID-552E7373-BF93-477D-9DA3-B2C9386F2877

### Expected Behaviour

While linting Oracle SQL from the command line, SQLFluff ought to be able to parse ALTER TABLE EXAMPLE MODIFY (modify_col_properties) statements using the oracle dialect.

### Observed Behaviour

L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'ALTER                                                                                                                                                                                                                               
                       | TABLE PRODUCT_INFORMATION MODIFY (...'
L:   2 | P:   1 |  PRS | Line 2, Position 1: Found unparsable section: 'ALTER
                       | TABLE PRODUCT_INFORMATION MODIFY (...'


### How to reproduce

Using the oracle dialect command line arg, run a SQL statement similar to:

```sql
ALTER TABLE PRODUCT_INFORMATION MODIFY (ID NOT NULL ENABLE);
ALTER TABLE PRODUCT_INFORMATION MODIFY (MIN_PRICE DEFAULT 10);
```

### Dialect

`oracle`

### Version

sqlfluff version `2.0.7`, Python version `3.7.3`

### Configuration

```
[sqlfluff]
verbose = 0
dialect = oracle
output_line_length = 120
encoding = utf-8
large_file_skip_byte_limit = 0
max_line_length = 120

[sqlfluff:indentation]
indent_unit = space
tab_space_size = 4

[sqlfluff:rules:capitalisation.keywords]
# Keywords
capitalisation_policy = upper
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
